app/src/main/java/com/example/habitsmasher/UserDatabaseHelper.java

```diff
@@ -0,0 +1,88 @@
+package com.example.habitsmasher;
+
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.List;
+
+public class UserDatabaseHelper {
+    private String _userID;
+    private final TextView _numberOfFollowers;
+    private final TextView _numberOfFollowing;
+
+    public UserDatabaseHelper(String userID, TextView numFollowers, TextView numFollowing) {
+        _numberOfFollowers = numFollowers;
+        _numberOfFollowing = numFollowing;
+        _userID = userID;
+    }
+
+    public void setFollowingCountOfUser() {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        DocumentReference userRef = db.collection("Users").document(_userID);
+
+        userRef.get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                if (task.isSuccessful()) {
+                    DocumentSnapshot userSnapshot = task.getResult();
+
+                    List<Object> following = (List<Object>) userSnapshot.get("following");
+
+                    setNumberOfFollowing(following);
+                }
+            }
+        });
+    }
+
+    public void setFollowerCountOfUser() {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        DocumentReference userRef = db.collection("Users").document(_userID);
+
+        userRef.get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                if (task.isSuccessful()) {
+                    DocumentSnapshot userSnapshot = task.getResult();
+
+                    List<Object> followers = (List<Object>) userSnapshot.get("followers");
+
+                    setNumberOfFollowers(followers);
+                }
+            }
+        });
+
+    }
+
+    private void setNumberOfFollowing(List<Object> following) {
+        if (following != null && !following.isEmpty()) {
+            if (following.get(0) == "" && !following.isEmpty()) {
+                _numberOfFollowing.setText("0");
+            } else {
+                _numberOfFollowing.setText(String.valueOf(following.size()));
+            }
+        } else {
+            _numberOfFollowing.setText("0");
+        }
```

Comment:
Although seemingly complex at first, couldn't find a better way to do this either. There are multiple cases to consider here... If the array isn't created for the given user in the database (they have never followed or been followed by any user), when fetching from the db we can get some NPEs. This is guarding against those.